internal/backends/postgresql/collection.go

```diff
@@ -245,8 +248,53 @@ func (c *collection) DeleteAll(ctx context.Context, params *backends.DeleteAllPa
 
 // Explain implements backends.Collection interface.
 func (c *collection) Explain(ctx context.Context, params *backends.ExplainParams) (*backends.ExplainResult, error) {
-	// TODO https://github.com/FerretDB/FerretDB/issues/3389
-	return new(backends.ExplainResult), nil
+	p, err := c.r.DatabaseGetExisting(ctx, c.dbName)
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	if p == nil {
+		return new(backends.ExplainResult), nil
+	}
+
+	meta, err := c.r.CollectionGet(ctx, c.dbName, c.name)
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	if meta == nil {
+		return &backends.ExplainResult{
+			QueryPlanner: must.NotFail(types.NewDocument()),
+		}, nil
+	}
+
+	// TODO https://github.com/FerretDB/FerretDB/issues/3414
+	q := fmt.Sprintf(
+		`EXPLAIN (VERBOSE true, FORMAT JSON) SELECT %s FROM %s`,
+		metadata.DefaultColumn,
+		pgx.Identifier{c.dbName, meta.TableName}.Sanitize(),
+	)
+
+	rows, err := p.Query(ctx, q)
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	iter := newQueryIterator(ctx, &iteratorParams{
+		rows:      rows,
+		unmarshal: unmarshalExplain,
+	})
```

Comment:
Let's not make change to `newQueryIterator(..)` func.

Let's do it similar to how it's done in sqlite to contain explain specific things within this scope https://github.com/FerretDB/FerretDB/blob/a29e4fb87d96025c9f7376603a313742b646df7e/internal/backends/sqlite/collection.go#L268

Reply:
Just clarifying https://github.com/FerretDB/FerretDB/pull/3465#discussion_r1343323383